File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s)!=len(t):
            return False
        
        table={}
        
        for i in range(len(s)):
            
            if s[i] not in table :
                table[s[i]]=1       
            else:
                table[s[i]]+=1
                
        for i in range(len(t)):
                        
            if t[i] not in table:
                table[t[i]]=-1       
            else:
                table[t[i]]-=1
                
        for key in table:
            
            if table[key]<0:
                return False
            
        return True
                
            

File: 242-valid-anagram/242-valid-anagram.go
func isAnagram(s string, t string) bool {
    
    if len(s) != len(t) {
        return false;
    }
    
    table := make(map[rune]int)
    
    for _, val := range s {
        table[val]++;
    }  
    
    for _, val := range t {
        table[val]--;
    }
    
    for idx, _ := range table {
        if table[idx] != 0 {
            return false;
        }
    }
    
    return true
}